docs/plugin-authoring-evaluator.md

```diff
@@ -0,0 +1,380 @@
+    const finalPrompt = promptTemplate({
+      output: d.output,
+    });
+
+    // Call the LLM to generate an evaluation result
+    const response = await generate({
+      model: judgeLlm,
+      prompt: finalPrompt,
+      config: judgeConfig,
+    });
+
+    // Parse the output
+    const parsedResponse = extractJson<DeliciousnessDetectionResponse>(
+      response.text()
+    );
+    if (!parsedResponse) {
+      throw new Error(`Unable to parse evaluator response: ${response.text()}`);
+    }
+
+    // Return a scored response
+    return {
+      score: parsedResponse.verdict,
+      details: { reasoning: parsedResponse.reason },
+    };
+  } catch (err) {
+    console.debug(
+      `Deliciousness evaluation failed with error ${err} for sample ${JSON.stringify(
+        d
+      )}`
+    );
+    throw err;
+  }
+}
+```
+
+#### Define the evaluator action
+
+The final step is to write a function that defines the evaluator action itself.
+
+```ts
+/**
+ * Create the Deliciousness evaluator action.
+ */
+export function createDeliciousnessEvaluator<
+  ModelCustomOptions extends z.ZodTypeAny,
+>(
+  judge: ModelReference<ModelCustomOptions>,
+  judgeConfig: z.infer<ModelCustomOptions>
+): EvaluatorAction {
+  return defineEvaluator(
+    {
+      name: `myAwesomeEval/deliciousness`,
+      displayName: 'Deliciousness',
+      definition: 'Determines if output is considered delicous.',
+    },
+    async (datapoint: BaseDataPoint) => {
+      const score = await deliciousnessScore(judge, datapoint, judgeConfig);
+      return {
+        testCaseId: datapoint.testCaseId,
+        evaluation: score,
+      };
+    }
+  );
+}
+```
+
+### Non-LLM-based Evaluators
+
+An non-LLM-based evaluator is any function used evaluate the input, context, or output of your genAI feature.
+
+Non-LLM-based evaluators in Genkit are made up of 2 components:
+
+- A scoring function
+- An evaluator action
+
+#### Define the scoring function
+
+Just like the LLM-based evaluator, define the scoring function. In this case, the scoring function does not need to know about the judge LLM or its config.
+
+```ts
+const US_PHONE_REGEX =
+  /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4}$/i;
+
+/**
+ * Scores whether an individual datapoint matches a US Phone Regex.
+ */
+export async function usPhoneRegexScore(
+  dataPoint: BaseDataPoint
+): Promise<Score> {
+  const d = dataPoint;
+  try {
+    if (!d.output || typeof d.output !== 'string') {
+      throw new Error('String output is required for regex matching');
+    }
+    const matches = US_PHONE_REGEX.test(d.output as string);
+    const reasoning = matches
+      ? `Output matched regex ${regex.source}`
+      : `Output did not match regex ${regex.source}`;
+    return {
+      score: matches,
+      details: { reasoning },
+    };
+  } catch (err) {
+    console.debug(
+      ` regex matcher failed with error ${err} for sample ${JSON.stringify(d)}`
+    );
+    throw err;
```

Comment:
Same- consider removing the log and re-throw.

Reply:
done.